storage/innobase/gis/gis0geo.cc

```diff
@@ -364,8 +364,9 @@ mbr_join_square(
 		b += 2;
 	} while (a != end);
 
-	/* Check for infinity or NaN, so we don't get NaN in calculations */
-	if (my_isinf(square) || my_isnan(square)) {
+	/* Check if finite (not infinity or NaN),
+	so we don't get NaN in calculations */
+	if (!finite(square)) {
```

Comment:
I think that the last line should say `!isfinite(square)` rather than `finite(square)`.
According to `man finite` on my Debian GNU/Linux system, `finite()` is a BSD function, while `isfinite()` is in C99.
We should consistently use the more portable function.